diff --git a/cmake/onnxruntime.cmake b/cmake/onnxruntime.cmake
index 010696a61..dfd6fd012 100644
--- a/cmake/onnxruntime.cmake
+++ b/cmake/onnxruntime.cmake
@@ -142,8 +142,16 @@ if(onnxruntime_BUILD_SHARED_LIB)
   endif()
   target_include_directories(onnxruntime PRIVATE ${ONNXRUNTIME_ROOT} PUBLIC "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/onnxruntime>")
 
+  target_compile_definitions(onnxruntime PRIVATE FILE_NAME=\"onnxruntime_sx$<$<CONFIG:Debug>:Debug>.dll\")
+  set_target_properties(onnxruntime PROPERTIES OUTPUT_NAME "onnxruntime_sx")
 
-  target_compile_definitions(onnxruntime PRIVATE FILE_NAME=\"onnxruntime.dll\")
+  if(MSVC)
+    add_link_options("/maxilksize:0x7ff00000")
+  endif()
+
+  if((CMAKE_BUILD_TYPE MATCHES "Debug") OR ("Debug" IN_LIST CMAKE_CONFIGURATION_TYPES))
+    add_compile_definitions(SX_DEBUG_BUILD)
+  endif()
 
   if(UNIX)
     if (APPLE)
@@ -289,7 +297,9 @@ if(NOT APPLE AND NOT WIN32)
       PUBLIC_HEADER "${ONNXRUNTIME_PUBLIC_HEADERS}"
       VERSION ${ORT_VERSION}
       SOVERSION 1
-      FOLDER "ONNXRuntime")
+      FOLDER "ONNXRuntime"
+      NO_SONAME 1 # AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+    )
   else()
     set_target_properties(onnxruntime PROPERTIES
       PUBLIC_HEADER "${ONNXRUNTIME_PUBLIC_HEADERS}"
diff --git a/onnxruntime/core/session/provider_bridge_ort.cc b/onnxruntime/core/session/provider_bridge_ort.cc
index 41cf8be1d..a4ed4dece 100644
--- a/onnxruntime/core/session/provider_bridge_ort.cc
+++ b/onnxruntime/core/session/provider_bridge_ort.cc
@@ -125,14 +125,26 @@ using EtwRegistrationManager_EtwInternalCallback = EtwRegistrationManager::EtwIn
 
 // The filename extension for a shared library is different per platform
 #ifdef _WIN32
-#define LIBRARY_PREFIX
-#define LIBRARY_EXTENSION ORT_TSTR(".dll")
+  #define LIBRARY_PREFIX
+  #ifdef SX_DEBUG_BUILD
+    #define LIBRARY_EXTENSION ORT_TSTR("d.dll")
+  #else
+    #define LIBRARY_EXTENSION ORT_TSTR(".dll")
+  #endif
 #elif defined(__APPLE__)
-#define LIBRARY_PREFIX "lib"
-#define LIBRARY_EXTENSION ".dylib"
+  #define LIBRARY_PREFIX "lib"
+  #ifdef SX_DEBUG_BUILD
+    #define LIBRARY_EXTENSION ORT_TSTR("d.dylib")
+  #else
+    #define LIBRARY_EXTENSION ORT_TSTR(".dylib")
+  #endif
 #else
-#define LIBRARY_PREFIX "lib"
-#define LIBRARY_EXTENSION ".so"
+  #define LIBRARY_PREFIX "lib"
+  #ifdef SX_DEBUG_BUILD
+    #define LIBRARY_EXTENSION ORT_TSTR("d.so")
+  #else
+    #define LIBRARY_EXTENSION ORT_TSTR(".so")
+  #endif
 #endif
 
 #ifdef __GNUC__
