diff --git a/cmake/onnxruntime.cmake b/cmake/onnxruntime.cmake
index e1d9810..8ee6a16 100644
--- a/cmake/onnxruntime.cmake
+++ b/cmake/onnxruntime.cmake
@@ -142,8 +142,12 @@ if(onnxruntime_BUILD_SHARED_LIB)
   endif()
   target_include_directories(onnxruntime PRIVATE ${ONNXRUNTIME_ROOT} PUBLIC "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/onnxruntime>")
 
+  target_compile_definitions(onnxruntime PRIVATE FILE_NAME=\"onnxruntime_sx$<$<CONFIG:Debug>:Debug>.dll\")
+  set_target_properties(onnxruntime PROPERTIES OUTPUT_NAME "onnxruntime_sx")
 
-  target_compile_definitions(onnxruntime PRIVATE FILE_NAME=\"onnxruntime.dll\")
+  if((CMAKE_BUILD_TYPE MATCHES "Debug") OR ("Debug" IN_LIST CMAKE_CONFIGURATION_TYPES))
+    add_compile_definitions(SX_DEBUG_BUILD)
+  endif()
 
   if(UNIX)
     if (APPLE)
@@ -287,15 +291,16 @@ if(NOT APPLE AND NOT WIN32)
   if(CMAKE_SYSTEM_NAME MATCHES "AIX")
     set_target_properties(onnxruntime PROPERTIES
       PUBLIC_HEADER "${ONNXRUNTIME_PUBLIC_HEADERS}"
-      VERSION ${ORT_VERSION}
-      SOVERSION 1
-      FOLDER "ONNXRuntime")
+      # VERSION ${ORT_VERSION}
+      # SOVERSION 1
+      FOLDER "ONNXRuntime"
+    )
   else()
     set_target_properties(onnxruntime PROPERTIES
       PUBLIC_HEADER "${ONNXRUNTIME_PUBLIC_HEADERS}"
       LINK_DEPENDS ${SYMBOL_FILE}
-      VERSION ${ORT_VERSION}
-      SOVERSION 1
+      # VERSION ${ORT_VERSION}
+      # SOVERSION 1
       FOLDER "ONNXRuntime")
   endif()
 else()
@@ -303,7 +308,7 @@ else()
   set_target_properties(onnxruntime PROPERTIES
     PUBLIC_HEADER "${ONNXRUNTIME_PUBLIC_HEADERS}"
     LINK_DEPENDS ${SYMBOL_FILE}
-    VERSION ${ORT_VERSION}
+    # VERSION ${ORT_VERSION}
     FOLDER "ONNXRuntime")
 endif()
 install(TARGETS onnxruntime
diff --git a/onnxruntime/core/session/provider_bridge_ort.cc b/onnxruntime/core/session/provider_bridge_ort.cc
index 19c636b..075ceff 100644
--- a/onnxruntime/core/session/provider_bridge_ort.cc
+++ b/onnxruntime/core/session/provider_bridge_ort.cc
@@ -125,14 +125,26 @@ using EtwRegistrationManager_EtwInternalCallback = EtwRegistrationManager::EtwIn
 
 // The filename extension for a shared library is different per platform
 #ifdef _WIN32
-#define LIBRARY_PREFIX
-#define LIBRARY_EXTENSION ORT_TSTR(".dll")
+  #define LIBRARY_PREFIX
+  #ifdef SX_DEBUG_BUILD
+    #define LIBRARY_EXTENSION ORT_TSTR("d.dll")
+  #else
+    #define LIBRARY_EXTENSION ORT_TSTR(".dll")
+  #endif
 #elif defined(__APPLE__)
-#define LIBRARY_PREFIX "lib"
-#define LIBRARY_EXTENSION ".dylib"
+  #define LIBRARY_PREFIX "lib"
+  #ifdef SX_DEBUG_BUILD
+    #define LIBRARY_EXTENSION ORT_TSTR("d.dylib")
+  #else
+    #define LIBRARY_EXTENSION ORT_TSTR(".dylib")
+  #endif
 #else
-#define LIBRARY_PREFIX "lib"
-#define LIBRARY_EXTENSION ".so"
+  #define LIBRARY_PREFIX "lib"
+  #ifdef SX_DEBUG_BUILD
+    #define LIBRARY_EXTENSION ORT_TSTR("d.so")
+  #else
+    #define LIBRARY_EXTENSION ORT_TSTR(".so")
+  #endif
 #endif
 
 #ifdef __GNUC__
