diff --git a/cmake/onnxruntime_providers_openvino.cmake b/cmake/onnxruntime_providers_openvino.cmake
index 5a831a106..d0593f0b6 100644
--- a/cmake/onnxruntime_providers_openvino.cmake
+++ b/cmake/onnxruntime_providers_openvino.cmake
@@ -46,6 +46,7 @@
 
   if(NOT MSVC)
     target_compile_options(onnxruntime_providers_openvino PRIVATE "-Wno-parentheses")
+    target_compile_options(onnxruntime_providers_openvino PRIVATE "-Wno-free-nonheap-object")
   endif()
   add_dependencies(onnxruntime_providers_openvino onnxruntime_providers_shared ${onnxruntime_EXTERNAL_DEPENDENCIES})
   target_include_directories(onnxruntime_providers_openvino SYSTEM PUBLIC ${ONNXRUNTIME_ROOT} ${CMAKE_CURRENT_BINARY_DIR} ${OpenVINO_INCLUDE_DIR} ${OPENVINO_INCLUDE_DIR_LIST} ${PYTHON_INCLUDE_DIRS} $ENV{OPENCL_INCS} $ENV{OPENCL_INCS}/../../cl_headers/)
@@ -88,3 +89,38 @@ set_target_properties(onnxruntime_providers_openvino PROPERTIES
   MAP_IMPORTED_CONFIG_RELEASE RelWithDebInfo
   MAP_IMPORTED_CONFIG_DEBUG RelWithDebInfo
   )
+
+# Install OpenVINO runtime libraries
+if(WIN32)
+  set(OPENVINO_LIB_EXTENSION "dll")
+else()
+  set(OPENVINO_LIB_EXTENSION "so")
+endif()
+
+string(STRIP $ENV{INTEL_OPENVINO_DIR} OPENVINO_PATH_RAW)
+cmake_path(SET OPENVINO_PATH ${OPENVINO_PATH_RAW})
+
+# Find OpenVINO runtime libraries
+file(GLOB OPENVINO_RUNTIME_LIB_FILES FOLLOW_SYMLINKS "${OPENVINO_PATH}/runtime/lib/intel64/Release/*.${OPENVINO_LIB_EXTENSION}")
+file(GLOB OPENVINO_RUNTIME_BIN_FILES FOLLOW_SYMLINKS "${OPENVINO_PATH}/runtime/bin/intel64/Release/*.${OPENVINO_LIB_EXTENSION}")
+
+# Find TBB libraries
+file(GLOB OPENVINO_TBB_LIB_FILES FOLLOW_SYMLINKS "${OPENVINO_PATH}/runtime/3rdparty/tbb/lib/*.${OPENVINO_LIB_EXTENSION}")
+file(GLOB OPENVINO_TBB_BIN_FILES FOLLOW_SYMLINKS "${OPENVINO_PATH}/runtime/3rdparty/tbb/bin/*.${OPENVINO_LIB_EXTENSION}")
+
+file(GLOB_RECURSE ALL_DAMN_OPENVINO_FILES LIST_DIRECTORIES true RELATIVE "${OPENVINO_PATH}" "${OPENVINO_PATH}/*")
+
+message(FATAL_ERROR "OPENVINO: '${OPENVINO_PATH}', '${OPENVINO_RUNTIME_LIB_FILES}', '${OPENVINO_RUNTIME_BIN_FILES}', '${OPENVINO_TBB_LIB_FILES}', '${OPENVINO_TBB_BIN_FILES}', '${ALL_DAMN_OPENVINO_FILES}'")
+
+# Install all found libraries
+if(OPENVINO_RUNTIME_LIB_FILES OR OPENVINO_RUNTIME_BIN_FILES OR OPENVINO_TBB_LIB_FILES OR OPENVINO_TBB_BIN_FILES)
+  install(
+    FILES
+      ${OPENVINO_RUNTIME_LIB_FILES}
+      ${OPENVINO_RUNTIME_BIN_FILES}
+      ${OPENVINO_TBB_LIB_FILES}
+      ${OPENVINO_TBB_BIN_FILES}
+    DESTINATION
+      ${CMAKE_INSTALL_LIBDIR}
+  )
+endif()
