diff --git a/cmake/onnxruntime_providers_cpu.cmake b/cmake/onnxruntime_providers_cpu.cmake
index a5f165dcb..31392e216 100644
--- a/cmake/onnxruntime_providers_cpu.cmake
+++ b/cmake/onnxruntime_providers_cpu.cmake
@@ -145,6 +145,8 @@ if (MSVC)
 #   if(NOT CMAKE_SIZEOF_VOID_P EQUAL 8)
 #      target_compile_options(onnxruntime_providers PRIVATE "/wd4244")
 #   endif()
+  target_compile_options(onnxruntime_providers PUBLIC "/maxilksize:0x7ff00000")
+  target_link_options(onnxruntime_providers PUBLIC "/maxilksize:0x7ff00000")
 endif()
 onnxruntime_add_include_to_target(onnxruntime_providers onnxruntime_common onnxruntime_framework onnx onnx_proto ${PROTOBUF_LIB} flatbuffers::flatbuffers Boost::mp11 safeint_interface)
 
